test/testdata/resolver/abstract_abstract.rb

```diff
@@ -0,0 +1,23 @@
+# typed: true
+
+class Parent
+  extend T::Sig
+  extend T::Helpers
+  abstract!
+
+  sig {abstract.returns(T.nilable(Integer))}
+  def example1; end
+
+  sig {abstract.returns(Integer)}
+  def example2; end
+end
+
+class Child < Parent
+  abstract!
+
+  sig {abstract.returns(Integer)}
+  def example1; end
+
+  sig {abstract.returns(T.nilable(Integer))}
+  def example2; end # error: Return type `T.nilable(Integer)` does not match return type of abstract method `Parent#example2`
```

Comment:
Do you think it would be possible/worth it to update this error message to convey that it is also allowed to narrow the type? Right now it gives the impression that the return type must be the exact same.

Reply:
It says that as an error note already:

<img width="882" alt="Screenshot 2023-04-03 at 3 54 53 PM" src="https://user-images.githubusercontent.com/5544532/229644048-eb735f60-8c0e-4755-b321-7fbac5b152bb.png">